src/dotnet/Telemetry.cs

```diff
@@ -19,6 +20,8 @@ public class Telemetry : ITelemetry
         private Dictionary<string, double> _commonMeasurements = null;
         private Task _trackEventTask = null;
 
+        internal static readonly string TelemetrySentinel  = Path.Combine(ApplicationEnvironment.ApplicationBasePath, $"{Product.Version}.dotnetTelemetry");
```

Comment:
NugetCache sentinel is used for creating the nuget cache? Hence the concurrent invocations are an issue, since it's a long operation. I think it's an overkill to reuse that for telemetry, moreover the file.exists guards on trying to create multiple entries, and the concurrency issue with file.create should be handled in file.create.